lib/FtpContext.js

```diff
@@ -6,21 +6,37 @@ const parseControlResponse = require("./parseControlResponse");
 /**
  * @typedef {Object} Task
  * @property {(...args: any[]) => void} resolve - Resolves the task.
- * @property {(...args: any[]) => void} reject - Rejects the task.
+ * @property {(err: Error) => void} reject - Rejects the task.
  */
 
 /**
- * @typedef {(response: Object, task: Task) => void} ResponseHandler
+ * @typedef {Object} FTPResponse
+ * @property {number} code - FTP response code
+ * @property {string} message - FTP response including code
  */
 
+/**
+ * @typedef {(err: Error | undefined, response: FTPResponse | undefined, task: Task) => void} ResponseHandler
+ */
+
+/**
+ * Describes an FTP server error response including the FTP response code.
+ */
+class FTPError extends Error {
+    get name() {
+        return this.constructor.name;
+    }
+}
+exports.FTPError = FTPError;
+
```

Comment:
I'm using `FTPError` just to mark an error that has been reported as one by the FTP server. That way, a task can differentiate between FTP response error and another socket error.